en/docs/guides/mfa/x509.md

```diff
@@ -0,0 +1,400 @@
+# Configure X509 Certificate Authenticator
+
+This page guides you through configuring the X509 certificate authenticator and the WSO2 Identity Server using a sample app to demonstrate authentication.
+
+---
+
+## Work with certificates
+
+X509 authentication requires the client to possess a Public Key Certificate (PKC). To know more about ublic Key Certificate (PKC) and Certificate Authority (CA), refer [Working with certificates](TO DO: Concepts)
+
+
+To create a sample certificate and create your own Certificate Authority to sign the certificates, follow the following steps:
+
+1.  The first step is to create the private RSA key:
+
+    ```
+    openssl genrsa -out rootCA.key 2048
+    ```
+
+    Here, the specified key size is 2048 bit. You can specify the key size for your private key.
+
+2.  Based on this key you can now generate an actual certificate which is valid for 10 years using the following command:
+
+    ```
+    openssl req -new -x509 -days 3650 -key rootCA.key -out rootCA.crt
+    ```
+
+3.  You are prompted to provide the following details, and the details you provide are incorporated into the certificate request. 
+    An example is shown below. Make sure you use the values that fit your use case.
+
+    - Country Name (2 letter code) [AU]: SL
+    - State or Province Name (full name) [Some-State]: Western
+    - Locality Name (eg, city) [ ]: Colombo
+    - Organization Name (eg, company) [Internet Widgits Pty Ltd]: WSO2
+    - Organizational Unit Name (eg, section) [ ]: QA
+    - Common Name (e.g. serverFQDN or YOUR name) [ ]: wso2is.com 
+    - Email Address [ ]: kim@wso2.com
+
+4.  An OpenSSL CA requires new files and supporting directories. Therefore, create a new directory.
+    Create the directory structure according to your `openssl.conf` format.
+
+    ```
+    mkdir -p demoCA/newcerts
+    ```
+
+5.  You also need some initial files inside your CA directory structure.
+
+    ```
+    touch demoCA/index.txt
+    echo '01' > demoCA/serial
+    ```
+
+6.  For the JVM to trust your newly created certificate import your certificate into your JVM trust store by executing the following command:
+
+    ```
+    keytool -import -noprompt -trustcacerts -alias rootCA -file rootCA.crt -keystore /Library/Java/JavaVirtualMachines/jdk1.8.0_191.jdk/Contents/Home/jre/lib/security/cacerts -storepass changeit
+    ```
+
+    !!! info "Got the 'permission denied' error?"
+        Note that when adding the certificate to the JVM trust store you may get the permission denied error. Running this command as an administrator resolves this permission issue. 
+
+        For example, if you are a Mac user, you can use sudo in front of this command to fix the permission issue.  
+
+7. Now you have created the CA to sign the certificate. To create the server certificate follow the steps given below:
+
+    1. Create the keystore that includes the private key by executing the following command:
```

Comment:
Use periods (.) at the end of sentences instead of colons (:)
 
Please change wherever applicable.